### ISSUE TITLE
Bounce Message Fails to Deliver

### ISSUE THREAD

**Issue by superman20** [haraka/Haraka#1907]
### system info

Haraka | Haraka.js - Version: 2.8.13
--- | :---
Node | v7.10.0
OS | Windows 7 (64-bit)

### Expected behavior

Bounce message to be delivered

### Observed behavior

Bounce message gets rejected due to invalid MAIL FROM command

### Steps to reproduce

Send Haraka an e-mail that is known to bounce.  When Haraka sends the bounce message back to the originating server it is rejected because of a bad MAIL FROM command.  Haraka will issue the following:
`[PROTOCOL] [...] [outbound] C: MAIL FROM:<>`
I have tracked this down to Line 2125 of file Outbound.js, function HMailItem.prototype.bounce_respond:
`var from = new Address ('<>');`

Shouldn't the from address be something like MAILER-DAEMON@doman.com?

Later edit: I do believe that MAIL FROM:<> is RFC valid, but I have found several SMTP servers that don't handle that well.  They seem to only function properly when there is a non-blank FROM command.  An example server is Charter's SMTP server at 68.114.188.69.

**Comment by baudehlo** [comment]
No absolutely not. You can put that in the From: header but in the SMTP conversation it should be <> as given.

**Comment by superman20** [comment]
After doing some research, it seems this is a touchy issue.  The RFC does recommend using MAIL FROM:<> for bounce messages.  It appears some SMTP server's don't accept it for various reasons.  I probed 15 random (large-scale) SMTP services and I found 4 servers who refused MAIL FROM:<>.  I'm sure my statistics aren't representative of the population, but it is alarming the number of legitimate bounces that will be never be delivered.  I'm going to close the issue as it's apparently a design decision and not a bug.
